engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -624,4 +625,15 @@ object WorkflowExecutionActor {
     Props(WorkflowExecutionActor(workflowDescriptor, serviceRegistryActor, jobStoreActor, subWorkflowStoreActor,
       callCacheReadActor, jobTokenDispenserActor, backendSingletonCollection, initializationData, restarting)).withDispatcher(EngineDispatcher)
   }
-}
\ No newline at end of file
+
+  implicit class EnhancedWorkflowOutputs(val outputs: Map[LocallyQualifiedName, WdlValue]) extends AnyVal {
+    def maxStringLength = 1000
+
+    def stripLarge = outputs map { case (k, v) =>
+      val wdlString = v.toWdlString
+
+      if (wdlString.length > maxStringLength) (k, WdlString(wdlString.take(maxStringLength) + " ... output too long, elided"))
+      else (k, v)
```

Comment:
Are you aware of `StringUtils.abbreviate`?  It doesn't do exactly what you have here but it might be a decent alternative.